substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -1872,6 +1882,33 @@ pub mod pallet {
 			ensure_signed(origin)?;
 			Self::do_payout_stakers_by_page(validator_stash, era, page)
 		}
+
+		/// Migrates an account's `RewardDestination::Controller` to
+		/// `RewardDestination::Account(controller)`.
+		///
+		/// Effects will be felt instantly (as soon as this function is completed successfully).
+		///
+		/// This will waive the transaction fee if the `payee` is successfully migrated.
+		#[pallet::call_index(27)]
+		#[pallet::weight(T::WeightInfo::update_payee())]
+		pub fn update_payee(
+			origin: OriginFor<T>,
+			controller: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			let _ = ensure_signed(origin)?;
+			let ledger = Self::ledger(StakingAccount::Controller(controller.clone()))?;
+
+			#[allow(deprecated)]
+			if Payee::<T>::get(&ledger.stash) != RewardDestination::Controller {
+				return Ok(Pays::Yes.into())
+			}
+
+			let _ = ledger
+				.set_payee(RewardDestination::Account(controller))
+				.defensive_proof("ledger was retrieved from storage, thus its bonded; qed.")?;
```

Comment:
This message seems more what I'd expect for an `expect()` to have. I'd add something that is more helpful when sieving through the debug logs.

```suggestion
				.defensive_proof("ledger should have been previously retrieved from storage.")?;
```